src/CommunityVoices/App/Api/Component/FileProcessor.php

```diff
@@ -0,0 +1,207 @@
+                       $columnName = $columnOrder[$i]; // XML requires no spaces
+                       $currentColumnData = $this->replaceTextInTags($data[$i]);
+                       if($columnName != "unrecognized") {
+                           $originalName = $sourceNameMapper[$columnName];
+                           if($columnName=="identifier") {
+                               $identifier = "i" . $this->cleanString($currentColumnData); // identifier must start with a letter for xml parse so this takes care of that
+                               if(!empty($identifier)) array_push($validIdentifiers["allIdentifiers"],["item" => $identifier]);
+                           } else {
+                               $dataToAdd['rowData'][$columnName] = ["originalName" => $originalName, "columnData" => $currentColumnData, "formattedName" => $columnName];
+                           }
+                       }
+                   }
+                   if($identifier) {
+                       $dataToAdd['rowData']["attribution"]["error"] = self::ERR_MISSING_ATTRIBUTION;
+                       $sheetData[$identifier] = $dataToAdd;
+                       $sheetData[$identifier]["quotes"] = []; // allows quotes related to source info
+                   }
+               }
+           }
+
+           fclose($f);
+        }
+
+         if (($f= fopen($quoteFilePath, "r")) !== FALSE) {
+             $columnOrder = []; // used to track column locations since we are going entirely by name instead of order
+             $givenColumnNames = fgetcsv($f);
+             foreach ($givenColumnNames as $column) {
+                 $formattedColumn = $this->cleanString($column);
+                 if(in_array($formattedColumn,$formattedQuoteNames)) {
+                     array_push($columnOrder,$formattedColumn);
+                 } else {
+                     array_push($columnOrder,"unrecognized");
+                     array_push($columnNameWarnings["unrecognized"],["item" => [$column]]);
+                 }
+             }
+
+
+             if(!in_array("contentcategory1",$columnOrder)) array_push($columnNameErrors,["item" => [self::ERR_NO_CONTENT_CATEGORIES]]);
+             if(!in_array("editedquotes",$columnOrder)) array_push($columnNameErrors,["item" => [self::ERR_NO_QUOTE]]);
+             if(!in_array("identifier",$columnOrder)) array_push($columnNameErrors,["item" => [self::ERR_NO_IDENTIFIER]]);
+             // These are both major errors. There is no need to parse the rest of the sheet as uploading will not be allowed if one of these errors occurs
+
+             for($i = 0; $i < count($formattedQuoteNames); $i++) {
+                 $formattedQuoteName = $formattedQuoteNames[$i];
+                 $unformattedQuoteName = self::BATCH_QUOTE_DATA[$i];
+                 if (!in_array($formattedQuoteName,$columnOrder))
+                   array_push($columnNameWarnings["expected"],["item" => [$unformattedQuoteName]]);
+             }
```

Comment:
Not necessary, but this could be patternized with the above similar for loop, only passing in the variables that are different.

Reply:
@Sammidysam I can look into this haha. I recall the logic being a little different in each loop but maybe I was wrong

Reply:
At this stage, it is probably not worth looking into, but I am me and had to mention it.